sale_commission_geo_assign_product_domain/wizards/wizard_geo_assign_partner.py

```diff
@@ -8,35 +8,40 @@ class WizardGeoAssign(models.TransientModel):
     _inherit = "wizard.geo.assign.partner"
 
     @api.model
-    def update_partner_data(self, partner, agent):
-        if agent.commission_type == "product_restricted":
-            for line in agent.commission_geo_group_ids:
-                if line.is_assignable(partner):
-                    partner.agent_ids = [(4, agent.id)]
-
-                    commission_item = partner.commission_item_agent_ids.filtered(
-                        lambda x: x.agent_id == agent
-                    )
-                    if commission_item:
-                        commission_item.group_ids = [
-                            (4, x) for x in line.mapped("commission_group_ids.id")
-                        ]
-                    else:
-                        partner.commission_item_agent_ids = [
-                            (
-                                0,
-                                0,
-                                {
-                                    "agent_id": agent.id,
-                                    "group_ids": [
-                                        (
-                                            6,
-                                            0,
-                                            line.mapped("commission_group_ids.id"),
-                                        )
-                                    ],
-                                },
-                            )
-                        ]
-        else:
-            super().update_partner_data(partner, agent)
+    def update_partner_data(self, partner, agents):
+        product_restricted_agents = agents.filtered(
+            lambda a: a.commission_type == "product_restricted"
+        )
+        for agent in product_restricted_agents:
+            assignable_lines = agent.commission_geo_group_ids.filtered(
+                lambda l: l.is_assignable(partner)
+            )
+            if not assignable_lines:
+                continue
+            partner.agent_ids = [(4, agent.id)]
+            for line in assignable_lines:
```

Comment:
**suggestion**(non-blocking): for an additional small performance gain, it'd be best to avoid the double loop (`filtered` loops once, and then the `for` loop is the second one) and manage it all in a single `for` loop and `if not line.is_assignable: continue`

Another small improvement: gather all the tuples you want to assign to partner.agent_ids (line 21) and assign them all at once at the end of the loop.